### PR TITLE
disposal tube refactor

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
you can count the rings on disposals code to see how old it is
this refactors the tube related code

## Why / Balance
i will be adding new disposal piece soon so time to clean up this mess

## Technical details
- general cleanup of tube/tagger/router code
- no virtual transit/junction component tagger and router are now separated from them
- sound params are on the sounds instead of hardcoded
- router event overriding is better
- make shared ui stuff less evil
- move router and tagger code into their own systems

## Media
router with matching tag

router without matching tag

junction still work yep

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
`SharedDisposalRouterComponent` removed and is now just the ui code in `DisposalRouterUi.cs`
`SharedDisposalTaggerComponent` removed and is now just the ui code in `DisposalTaggerUi.cs`
anything that has `DisposalRouterComponent` now also needs `DisposalJunctionComponent`
anything that has `DisposalTaggerComponent` now also needs `DisposalTransitComponent`

most likely nobodys made their own disposals pieces yet but who knows...

**Changelog**
no cl no fun

### CODE REVIEW

**Comment by brainfood1183** [comment]
Any chance u could fix the disposal unit sprite layers? it uses a layer called charging that isnt listed in the proto (must be in the component or something). If you try to create a new unit it will keep blinking back to the disposal_charging layer. Only way i found to fix this is to add a new unit sprites in a separate folder and you need to name the charging layer disposal_charging.

**Comment by metalgearsloth** [comment]
pro

File: Content.Server/Disposal/Tube/Systems/DisposalTaggerSystem.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        SubscribeLocalEvent<DisposalTaggerComponent, GetDisposalsNextDirectionEvent>(OnGetNextDirection);
+        // TODO: maybe this could use simple dialog thing that rename trick uses?
+        SubscribeLocalEvent<DisposalTaggerComponent, ActivatableUIOpenAttemptEvent>(OnOpenUIAttempt);
+        SubscribeLocalEvent<DisposalTaggerComponent, TaggerSetTagMessage>(OnSetTag);
+    }
+
+    private void OnComponentRemove(EntityUid uid, DisposalTaggerComponent comp, ComponentRemove args)
+    {
+        _ui.TryCloseAll(uid, DisposalTaggerUiKey.Key);
+    }
```
Comment (metalgearsloth) [inline Content.Server/Disposal/Tube/Systems/DisposalTaggerSystem.cs:37]: Same here.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Apparently I started a review at some point.

**Comment by deltanedas** [comment]
bruh i thought i deleted this :trollface:

**Pull request closed without merging**